instance/models/instance.py

```diff
@@ -509,11 +550,26 @@ def default_fork(self):
         return settings.DEFAULT_FORK
 
     @property
-    def reference_name(self):
+    def ephemeral_databases(self):
         """
-        A descriptive name for the instance, which includes meaningful attributes
+        Will this instance's databases disappear when the instance is
+        reprovisioned?
         """
-        return '{s.github_organization_name}/{s.branch_name} ({s.commit_short_id})'.format(s=self)
+        return not self.database_url and not self.mongo_url
+
+    @ephemeral_databases.setter
+    def ephemeral_databases(self, value):
+        """
+        Set this to True to unset any external database settings
+        """
+        if value is True:
+            self.database_url = ''
+            self.mongo_url = ''
+        elif value is False:
+            if self.ephemeral_databases is not False:
+                raise ValueError('Set database_url or mongo_url')
+        else:
+            raise TypeError('Expected bool')
```

Comment:
Making a dependent property writable strikes me as a somewhat surprising interface.  Given that this setting can only be meaningfully changed from `False` to `True`, I think it would be clearer to simply set the attributes this depends on, i.e. `database_url` and `mongo_url`, directly.